feat(#1596): Showcase examples in playground

sl-slaing opened the pull request:
### Description
Update Playground to use example profiles, and show more appropriate example by default.

### Changes
- Introduce a list of curated example profiles (examples.json)
- Update playground to show the examples
- Remove initial profile content (replace with first loaded example)

### Issue
Resolves #1596

sl-slaing commented:
@ColinEberhardt demonstrated here.
https://sl-slaing.github.io/datahelix/playground

ColinEberhardt commented on playground/index.html:
```diff
@@ -19,6 +19,10 @@
 <nav class="navbar navbar-expand-lg navbar-light bg-light justify-content-between">
   <a class="navbar-brand" href="..">DataHelix Generator</a>
   <div>
+    <button id="examplesButton" type="button" class="btn btn-outline-success">
```
I'd recommend using a drop down button: https://getbootstrap.com/docs/4.0/components/dropdowns/#single-button-dropdowns

ColinEberhardt commented on playground/index.js:
```diff
@@ -105,4 +108,84 @@ runButton.addEventListener("click", () => {
+  return listElement;
+}
+
+const showExamples = (show) => {
```
(minor) you don't need braces here

ColinEberhardt commented on playground/index.js:
```diff
@@ -105,4 +108,84 @@ runButton.addEventListener("click", () => {
+    : "none";
+}
+
+const hideExamples = () => {
```
again - no braces needed

ColinEberhardt commented on playground/index.js:
```diff
@@ -105,4 +108,84 @@ runButton.addEventListener("click", () => {
+    editor.setValue("Loading examples...");
+    hideExamples();
+
+    resonse.json().then(profileExamples => {
```
(minor)resonse

ColinEberhardt commented on playground/style.css:
```diff
@@ -47,3 +47,41 @@ nav .btn {
   height: 100% !important;
 }
+
+.examples-display {
```
rather than doing all your own styling, can you use the bootstrap drop-down? https://getbootstrap.com/docs/4.0/components/dropdowns/

It looks like it is relatively easy to turn this into a multi-column layout: https://codepen.io/gdeichen/pen/GrEaya

sl-slaing commented:
> generally looks good to me - a few comments on the implementation.
> 
> I also spotted that [this example](https://sl-slaing.github.io/datahelix/playground/#ewogICJmaWVsZHMiOiBbCiAgICB7CiAgICAgICJuYW1lIjogIndlZWtkYXlzIiwKICAgICAgInR5cGUiOiAic3RyaW5nIiwKICAgICAgIm51bGxhYmxlIjogZmFsc2UKICAgIH0sCiAgICB7CiAgICAgICJuYW1lIjogInZlZ2V0YWJsZXMiLAogICAgICAidHlwZSI6ICJzdHJpbmciLAogICAgICAibnVsbGFibGUiOiBmYWxzZQogICAgfSwKICAgIHsKICAgICAgIm5hbWUiOiAiZGlzY291bnQiLAogICAgICAidHlwZSI6ICJkZWNpbWFsIiwKICAgICAgIm51bGxhYmxlIjogdHJ1ZQogICAgfQogIF0sCiAgImNvbnN0cmFpbnRzIjogWwogICAgewogICAgICAiZmllbGQiOiAid2Vla2RheXMiLAogICAgICAiaW5TZXQiOiBbCiAgICAgICAgIk1vbmRheSIsCiAgICAgICAgIlR1ZXNkYXkiLAogICAgICAgICJXZWRuZXNkYXkiLAogICAgICAgICJUaHVyc2RheSIsCiAgICAgICAgIkZyaWRheSIKICAgICAgXQogICAgfSwKICAgIHsKICAgICAgImZpZWxkIjogInZlZ2V0YWJsZXMiLAogICAgICAiaW5TZXQiOiAidmVnZXRhYmxlcy5jc3YiCiAgICB9LAogICAgewogICAgICAiZmllbGQiOiAiZGlzY291bnQiLAogICAgICAiaW5TZXQiOiAicmVkdWN0aW9ucy5jc3YiCiAgICB9CiAgXQp9) doesn't work.
> 
> (edit - odd, that link didn't work? the 'in set' example is broken)

I think I've addressed all of your comments.

Regarding the not-working example, I'll take a look at this in a separate PR, as there are a couple of other examples that could do with a bit of a tidy-up.

sl-slaing commented:
There is an issue with this PR - it ignores the hash in the URL.

sl-slaing commented:
The example you've linked to as not working (inSet) won't work in the playground as it requires there to be additional files available during the generation - namely vegetables.csv and reductions.csv.

I'll get this example removed from the playground, together with any others that require additional resources.

sl-slaing commented:
@ColinEberhardt I've made a number of changes to the playground and think this PR is ready to merge - could you take a look again please.

There is another PR to resolve an issue with the 'inSet' example.

I'll have a look to see if it is possible to reinstate some of the removed examples through other means as a follow-up task

ColinEberhardt commented on playground/index.js:
```diff
@@ -5,15 +5,15 @@ const profileArea = document.getElementById("profile");
 const spinner = document.getElementById("spinner");
 const shareUrl = document.getElementById("shareUrl");
+const examplesDisplay = document.getElementById("examples");
+const examplesButton = document.getElementById("examplesButton");
```
is this unused?

ColinEberhardt commented on playground/index.html:
```diff
@@ -14,11 +14,16 @@
   <script src="https://unpkg.com/whatwg-fetch@3.0.0/dist/fetch.umd.js"></script>
   <script src="https://unpkg.com/popper.js@1.16.0/dist/umd/popper.min.js"></script>
   <script src="https://unpkg.com/bootstrap@4.4.0/dist/js/bootstrap.min.js"></script>
+  <script src="https://unpkg.com/showdown/dist/showdown.min.js"></script>
```
I'd link to a specific version (https://unpkg.com/showdown@1.9.1/dist/showdown.min.js) otherwise this might break in future!

ColinEberhardt commented on playground/style.css:
```diff
@@ -47,3 +47,31 @@ nav .btn {
   height: 100% !important;
 }
+
+#examples {
```
I'd prefer to just use the built-in styling for drop downs :-)
